### PR TITLE
Filter invalid mimetypes in boefjes output.

### PR DESCRIPTION
**Pull request by underdarknl**
### Changes
Filters out mimetypes that are too short, or don't contain a slash

### Issue link
closes https://github.com/minvws/nl-kat-coordination/issues/1456

### Code Checklist
- [x] All the commits in this PR are properly PGP-signed and verified.
- [ ] This PR only contains functionality relevant to the issue; tickets have been created for newly discovered issues.
- [ ] I have written unit tests for the changes or fixes I made.
- [ ] For any non-trivial functionality, I have added integration and/or end-to-end tests.
- [ ] I have performed a self-review of my code and refactored it to the best of my abilities.

### Communication
- [ ] I have informed others of any required `.env` changes files if required and changed the `.env-dist` accordingly.
- [ ] I have made corresponding changes to the documentation, if necessary.
- [ ] I have included comments in the code to elaborate on what is not self-evident from the code itself, including references to issues and discussions online, or implicit behavior of an interface.

---
## Checklist for code reviewers:
Copy-paste the checklist from [the docs/source/templates folder](https://github.com/minvws/nl-kat-coordination/blob/main/docs/source/templates/pull_request_template_review_code.md) into your comment.

---
## Checklist for QA:
Copy-paste the checklist from [the docs/source/templates folder](https://github.com/minvws/nl-kat-coordination/blob/main/docs/source/templates/pull_request_template_review_qa.md) into your comment.

### CODE REVIEW

**Comment by dekkers** [comment]
Wouldn't it be better to have the boefje fail instead of ignoring the mimetype?

**Comment by underdarknl** [comment]
> Wouldn't it be better to have the boefje fail instead of ignoring the mimetype?

Not all returned mimetypes might be invalid. And not all mime-types might be hard coded in the boefje (think http-headers being used to fill mime-types). Throwing a warning for invalid mime-types seems reasonable to me, as it might still allow other normalizers to ingest the data based on for example the boefje/$boefjename mimetype. the mime-types for each raw file in a job are visible in the task-list. but maybe we should find a way to also show any warnings there.

**Comment by dekkers** [comment]
> Not all returned mimetypes might be invalid. And not all mime-types might be hard coded in the boefje (think http-headers being used to fill mime-types). Throwing a warning for invalid mime-types seems reasonable to me, as it might still allow other normalizers to ingest the data based on for example the boefje/$boefjename mimetype. the mime-types for each raw file in a job are visible in the task-list. but maybe we should find a way to also show any warnings there.

All mimetypes are currently hardcoded/whitelisted as far as I know. I also don't think that creating an opportunity for attackers to inject raw files with arbitrary mimetypes is a good idea. I actually just realised that we fixed the `produces` field of boefje definition a few months ago, so we can simplify this by checking if the mimetype is listed in the `produces` of the boefje definition and give an error if that is not the case.

**Comment by underdarknl** [comment]
> > Not all returned mimetypes might be invalid. And not all mime-types might be hard coded in the boefje (think http-headers being used to fill mime-types). Throwing a warning for invalid mime-types seems reasonable to me, as it might still allow other normalizers to ingest the data based on for example the boefje/$boefjename mimetype. the mime-types for each raw file in a job are visible in the task-list. but maybe we should find a way to also show any warnings there.
> 
> All mimetypes are currently hardcoded/whitelisted as far as I know. I also don't think that creating an opportunity for attackers to inject raw files with arbitrary mimetypes is a good idea. I actually just realised that we fixed the `produces` field of boefje definition a few months ago, so we can simplify this by checking if the mimetype is listed in the `produces` of the boefje definition and give an error if that is not the case.

The produces list was never intended to be a complete list of all mime-types possible for a boefje. The whitelisting is valid though, and this is why we have/had a whitelist of mimetypes in for example the download boefje which makes sure a remote webserver cannot inject any 'special' mimetypes into openkat. (boefje/$boefje, or info/* error/* being the special ones).

We can start seeing the produces list as each boefje's whitelist. It would make things bit more strict, which is good for security (enforcing this would move to the runner, not the boefje itself), but we would still need to handle the special mime-types for errors and info message etc inside the boefje.

File: boefjes/boefjes/job_handler.py [hunk 1]
```diff
@@ -23,6 +23,8 @@
 from octopoes.models.exception import ObjectNotFoundException
 from octopoes.models.types import OOIType
 
+MIMETYPE_MIN_LENGTH = 5  # two chars before, and 2 chars after the slash ought to be reasonable
```
Comment (ammar92) [inline boefjes/boefjes/job_handler.py:26]: Although the suggested implementation works and seems as a practical approach, I was just curious if this is the right approach to validate MIME types. Why not just validate against the format `type/subtype;parameter=value`? A MIME type always has both a type and a subtype, never just one or the other, and an optional parameter can be added to provide additional details.

In my view a MIME type in the format `<two characters> / <two characters>` is not a more valid or reasonable MIME type than a non-existing or defined (with valid format) MIME type or a bogus one (e.g. `ab/cd`)

**Review by dekkers — APPROVED** [review]

**Pull request merged**